File: Baekjoon/Baekjoon15565.cpp
// Baekjoon15565.cpp
#include <iostream>

using namespace std;

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int n, k;
    cin >> n >> k;

    int dolls[1'000'000];
    for (int i = 0; i < n; i++)
        cin >> dolls[i];

    int ans = 0x3f3f3f3f, leftIdx = 0, currLion = 0;
    for (int rightIdx = 0; rightIdx < n; rightIdx++) {
        if (dolls[rightIdx] == 1)
            currLion++;

        while (currLion >= k) {
            ans = min(ans, rightIdx - leftIdx + 1);
            if (dolls[leftIdx] == 1)
                currLion--;
            leftIdx++;
        }
    }

    cout << (ans == 0x3f3f3f3f ? -1 : ans) << '\n';

    return 0;
}

File: Baekjoon/Baekjoon15565.py
# Baekjoon15565.py
import sys
input = sys.stdin.readline

INF = 10 ** 9

n, k = map(int, input().split())
dolls = list(map(int, input().split()))

ans = INF
leftIdx, currLion = 0, 0
for rightIdx in range(n):
	if (dolls[rightIdx] == 1):
		currLion += 1

	while (currLion >= k):
		ans = min(ans, rightIdx - leftIdx + 1)
		if (dolls[leftIdx] == 1):
			currLion -= 1
		leftIdx += 1

print(ans if ans != INF else -1)